homeassistant/components/maxcube/binary_sensor.py

```diff
@@ -32,6 +33,15 @@ def setup_platform(
         add_entities(devices)
 
 
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up a binary sensor for maxcube."""
+    setup_platform(hass, config_entry.data.copy(), async_add_devices)
```

Comment:
The `async_add_entities` logic is typically done here, not in `setup_platform`. Please use something like this for the above method:
```    hass.async_create_task(
        hass.config_entries.flow.async_init(
            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
        )
    )

Reply:
I did the change but something is wrong. I have the configuration in YML and removed configuration from UI, after restart binary  sensors do not work. `/config/entities` page shows the binary_sensors, but their status is "restored". I didn't do this change in `climate.py` and these entities are working fine. Have I missed something?

d2b56cd1a04482b4eb3c6b2cb0bd1afda3432288

Reply:
ok, lots of testing and i'm now sure I got it right - a8d7f62ef0ee0cda611e7642b5e3645bae5ff19e